src/analysis/retail/priest/shadow/modules/talents/Voidweaver/EntropicRift.tsx

```diff
@@ -41,22 +76,163 @@ class EntropicRift extends Analyzer {
     this.damageRift += event.amount + (event.absorbed || 0);
   }
 
+  onDP() {
+    if (this.selectedCombatant.hasBuff(SPELLS.SHADOW_PRIEST_VOIDWEAVER_ENTROPIC_RIFT_BUFF)) {
+      this.currentDP += 1;
+    }
+  }
+
+  onVB() {
+    if (this.selectedCombatant.hasBuff(SPELLS.SHADOW_PRIEST_VOIDWEAVER_ENTROPIC_RIFT_BUFF)) {
+      this.currentVB += 1;
+    }
+  }
+
   onCollapse(event: DamageEvent) {
     this.damageCollapse += event.amount + (event.absorbed || 0);
+
+    this.damageAmp = calculateEffectiveDamage(
+      event,
+      this.currentDP * COLLAPSING_VOID_DEVOURING_PLAGUE_MULTIPLIER,
+    );
+    this.damageAmpTotal += this.damageAmp;
+
+    //Since the collapse hits multiple targets, each collapse has multiple damage events but we only want one per cast.
+    if (event.timestamp - this.castTime >= 100) {
+      this.castTime = event.timestamp;
+
+      //Collapsing void
+      this.totalDP += this.currentDP;
+      this.finalizeRiftDamage();
+      this.currentDP = 0;
+
+      //Darkening Horizon
+      if (this.currentVB > 3) {
+        this.currentVB = 3;
+      } //VB can only give 3 seconds of extension
+      this.totalVB += this.currentVB;
+      this.finalizeRiftTime();
+      this.currentVB = 0;
+    }
```

Comment:
this probably works fine. if you run into timestamp issues, you might try an `EventLink` to connect the damage events to the cast and process the damage events linked from the cast event directly. don't need to go out and rewrite working code for this PR though

Reply:
Right now the timestamps are all at the same time, and there can never be more than one rift out a a time.  But in the future I'll change it to use EventLinks so it works better in case that ever changes.